akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -579,7 +581,10 @@ class HttpExt(private val config: Config)(implicit val system: ActorSystem) exte
         val fastFuture = FastFuture.successful(gateway)
         hostPoolCache.put(setup, fastFuture) // optimize subsequent gateway accesses
         gatewayPromise.success(gateway) // satisfy everyone who got a hold of our promise while we were starting up
-        whenShuttingDown.future.onComplete(_ ⇒ hostPoolCache.remove(setup, fastFuture))(fm.executionContext)
+        whenShuttingDown.future.onComplete { _ ⇒
+          hostPoolCache.remove(setup, fastFuture)
```

Comment:
I got it. My only grudge is this extra Promise. Thinking if we can simplify this. For example this code can be factored as:
```
val completelyCompletedFuture = whenShuttingDown.future.andThen {
  hostPoolCache.remove(setup, fastFuture)
}
```

but it might be that we cannot return a Future only inject a Promise.